client/src/containers/AuthContainer.test.tsx

```diff
@@ -98,3 +88,122 @@ describe("verifying valid input for signup page", () => {
     });
   });
 });
+
+describe("signUp component", () => {
+  describe("handleSubmit function", () => {
+    it("should show password mismatch alert", async () => {
+      screen.getByTestId("SignUpTabButton").click();
+
+      const fields = [
+        ["passwordForm", "password"],
+        ["confirmPassForm", "passwordd"],
+      ];
+      fields.forEach((pair) => {
+        fireEvent.change(
+          screen
+            .getByTestId(pair[0])
+            .querySelector("input") as HTMLInputElement,
+          { target: { value: pair[1] } }
+        );
+      });
+
+      const form = screen.getByTestId("form");
+      const mockAlert = jest
+        .spyOn(window, "alert")
+        .mockImplementation(() => {});
+
+      fireEvent.submit(form);
+      expect(mockAlert).toBeCalledWith("Passwords do not match.");
```

Comment:
You can assert that the mockApi function was not called rather than asserting that the alert was called with some error. If the error message changes for example, that would cause the test to fail when it really shouldn't (since there error message is just a visual thing, whereas this is a logic test).